eos-fix-home-dir-permissions

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -e
+
+# Prior to 3.1.6, home directory permissions were created with dir mode 755.
+# This is now fixed, but on upgrade from an earlier version to 3.1.6,
+# let's perform a one-time clean up of any existing home dirs.
+
+stamp_file=/var/eos-dir-mode-700
```

Comment:
Agreed with /var/lib/misc.  I was following the lead of other files that we had in /var, but yes it makes sense to stop polluting this top level directory.

I actually considered checking the permissions of /home/shared, though in theory an admin could decide to change those after the fact (as maybe they actually want the shared user's documents available to other users).  So, I think a stamp file is better -- plus it makes for a simple check that the service turns into a no-op.  (Is there a way for the service to check based on permissions and turn it into a no-op?)